Repositories/TeacherRepository.cs

```diff
@@ -73,7 +76,7 @@ public TeacherDto GetTeacher(int teacherId)
         {
             var user = _context.Teachers
                 .Where(t => t.Id == teacherId)
-                .Select(t => t.User).First();
+                .Select(t => t.User).FirstOrDefault();
```

Comment:
better to leave as is or add a check on the case when teacher is no longer a teacher (user == null)